ios/Classes/SwiftOptimoveFlutterSdkPlugin.swift

```diff
@@ -0,0 +1,366 @@
+        flutterEventChannel.setStreamHandler(self)
+        self.initOptimove(from: optimoveKeys)
+    }
+    
+    fileprivate func initOptimove(from optimoveKeys: OptimoveKeys){
+        let config = OptimoveConfigBuilder(optimoveCredentials: optimoveKeys.optimoveCredentials, optimobileCredentials: optimoveKeys.optimobileCredentials)
+        
+        if (optimoveKeys.enableDeferredDeepLinking) {
+            let dlHandler: DeepLinkHandler = { deepLinkResolution in
+                self.emitDeeplinkResolved(deepLinkResolution: deepLinkResolution)
+            }
+            if let cname = optimoveKeys.cname {
+                config.enableDeepLinking(cname: cname, dlHandler)
+            } else {
+                config.enableDeepLinking(dlHandler)
+            }
+        }
+        
+        if #available(iOS 10, *) {
+            config.setPushReceivedInForegroundHandler(pushReceivedInForegroundHandlerBlock: { notification , UNNotificationPresentationOptions -> Void in
+                self.emitPushNotificationReceivedEvent(pushNotification: notification)
+            })
+        }
+        
+        config.setPushOpenedHandler(pushOpenedHandlerBlock: { notification in
+            self.emitPushNotificationOpenedEvent(pushNotification: notification)
+        })
+        
+        if optimoveKeys.inAppConsentStrategy != .disabled {
+            config.enableInAppMessaging(inAppConsentStrategy: optimoveKeys.inAppConsentStrategy == .autoEnroll ? .autoEnroll : .explicitByUser)
+        }
+        
+        config.setInAppDeepLinkHandler { inAppButtonPress in
+            self.emitInappButtonPress(inAppButtonPress: inAppButtonPress)
+        }
+
+        Optimove.initialize(with: config.build())
+        setAdditionalListeners()
+    }
+    
+    fileprivate func setAdditionalListeners(){
+        OptimoveInApp.setOnInboxUpdated {
+            self.emitInboxUpdated()
+        }
+    }
+        
+    public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
+        switch call.method {
+            case "registerUser":
+              Optimove.registerUser(sdkId: (call.arguments as! Dictionary<String, Any>)["userId"] as! String, email: (call.arguments as! Dictionary<String, Any>)["email"] as! String)
+              result(nil)
+            case "setUserId":
+              Optimove.setUserId((call.arguments as! Dictionary<String, Any>)["userId"] as! String)
+              result(nil)
+            case "setUserEmail":
+              Optimove.setUserEmail(email: (call.arguments as! Dictionary<String, Any>)["email"] as! String)
+              result(nil)
+            case "reportEvent":
+              handleReportEvent(call)
+              result(nil)
+            case "reportScreenVisit":
+              handleReportScreenVisit(call)
+              result(nil)
+            case "getUserId":
+              result(nil)
```

Comment:
TODO?

Reply:
The native iOS SDK doesnt expose this yet. We could remove this case so that calling this method from Flutter would end in the default block but I think we better return nil explicitly for now

Reply:
I think a preference would be to remove it for now from both platforms.

Inconsistent behaviour isn't ideal in x-plat SDKs, and we have the additional complication of this value being the originalVisitorId _or_ user ID. Probably better to remove and add later if we have a need.

Reply:
Removed this API completely